modules/widgets/simple-payments.php

```diff
@@ -0,0 +1,330 @@
+		// TODO: This would be nice, but I don't know how this function works.
+		// if ( is_active_widget( false, false, $this->id_base, true ) ) {
+			wp_enqueue_style( 'simple-payments-widget', self::$url . '/simple-payments/style.css', array() );
+		// }
+	}
+
+	protected static function get_product_defaults() {
+		if ( ! isset( self::$product_defaults ) ) {
+			$products = get_posts( array(
+				'numberposts' => 1,
+				'orderby' => 'date',
+				'post_type' => Jetpack_Simple_Payments::$post_type_product,
+			) );
+			$current_user = wp_get_current_user();
+			$default_email = $current_user->user_email;
+			$default_multiple = '0';
+			$default_currency = 'USD';
+			// Get the default email, currency and "multiple" flag from the last product created, for user's convenience
+			if ( ! empty( $products ) ) {
+				$default_email = get_post_meta( $products[0]->ID, 'spay_email', true );
+				$default_multiple = get_post_meta( $products[0]->ID, 'spay_multiple', true );
+				$default_currency = get_post_meta( $products[0]->ID, 'spay_currency', true );
+			}
+			self::$product_defaults = array( $default_email, $default_multiple, $default_currency );
+		}
+		return self::$product_defaults;
+	}
+
+	protected function get_product_args( $product_id ) {
+		$product = $product_id ? get_post( $product_id ) : null;
+		$product_args = array();
+		if ( $product && ! is_wp_error( $product ) && $product->post_type === Jetpack_Simple_Payments::$post_type_product ) {
+			$product_args = array(
+				'name' => get_the_title( $product ),
+				'description' => $product->post_content,
+				'currency' => get_post_meta( $product->ID, 'spay_currency', true ),
+				'price' => get_post_meta( $product->ID, 'spay_price', true ),
+				'multiple' => get_post_meta( $product->ID, 'spay_multiple', true ),
+				'email' => get_post_meta( $product->ID, 'spay_email', true ),
+			);
+		}
+
+		$product_args = wp_parse_args( $product_args, array(
+			'name' => '',
+			'description' => '',
+			'currency' => null,
+			'price' => '',
+			'multiple' => null,
+			'email' => null,
+		) );
+
+		if ( isset( $product_args['currency'] ) && isset( $product_args['multiple'] ) && isset( $product_args['email'] ) ) {
+			// All fields are present, no need to "fill the blanks"
+			return $product_args;
+		}
+
+		list( $default_email, $default_multiple, $default_currency ) = self::get_product_defaults();
+		if ( ! isset( $product_args['email'] ) ) {
+			$product_args['email'] = $default_email;
+		}
+		if ( ! isset( $product_args['multiple'] ) ) {
+			$product_args['multiple'] = $default_multiple;
+		}
+		if ( ! isset( $product_args['currency'] ) ) {
+			$product_args['currency'] = $default_currency;
+		}
+		return $product_args;
+	}
+
+    /**
+     * Widget
+     */
+    function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, array(
+			'title' => '',
+			'product_id' => null,
+		) );
+
+		echo $args['before_widget'];
+
+		$title = apply_filters( 'widget_title', $instance['title'] );
```

Comment:
Could you add a docblock here, to make our parser happy? This should work:
`/** This filter is documented in core/src/wp-includes/default-widgets.php */`